File: exams/fulvipes-basic-exam/hero-battle/03-HeroBattle/03-HeroBattle/BaseHero.cs
using System;

namespace _03_HeroBattle
{
    public abstract class BaseHero
    {
        protected string name;
        protected double motivation;
        protected string heroUniverse;

        public BaseHero(String name)
        {
            this.name = name;
            heroUniverse = GetType().Name;
        }
        
        public abstract void Punch(IPunchable other);

        public abstract int GetMotivationLevel();

        public string GetName()
        {
            return name;
        }

        public string GetHeroUniverse()
        {
            return heroUniverse;
        }
    }
}

File: exams/fulvipes-basic-exam/hero-battle/BaseHero.ts
import Punchable from './Punchable';

export default abstract class BaseHero {

  private name: string;

  constructor(name: string) {
    this.name = name;
  }

  public abstract punch(other: Punchable): void

  public abstract getMotivationLevel(): number
  
  public abstract toString(): string

  public getName(): string {
    return this.name;
  }
}
